src/upgrade/upgrade_scripts/5.10.0/upgrade_bucket_lifecycle_structure.js

```diff
@@ -0,0 +1,47 @@
+/* Copyright (C) 2016 NooBaa */
+"use strict";
+
+const util = require('util');
+
+async function run({ dbg, system_store, system_server }) {
+
+    try {
+        dbg.log0('Starting upgrade lifecycle...');
+        let new_rules;
+        const buckets = [];
+
+        for (const bucket of system_store.data.buckets) {
+            //Do not update if there are no lifecycle_configuration_rules.
+            if (!bucket.lifecycle_configuration_rules) return;
+
+            new_rules = bucket.lifecycle_configuration_rules.map(lifecycle_configuration_rules => ({
+                id: lifecycle_configuration_rules.id,
+                filter: { prefix: lifecycle_configuration_rules.prefix },
+                status: lifecycle_configuration_rules.status,
+                expiration: lifecycle_configuration_rules.expiration,
+            }));
+
+            buckets.push({
+                _id: bucket._id,
+                lifecycle_configuration_rules: new_rules,
+            });
+        }
+
+        if (new_rules.length > 0) {
```

Comment:
Undefined new_rules will break your code. Check buckets length instead and then move new_rules declaration to inside the for.